### PR TITLE
Use operator by default + daemonset to operator migration

### PR DESCRIPTION
**Pull request by haibinxie**
Known issues:
1. STC is not deleted with helm delete after migration, tracked by [OPERATOR-740](https://portworx.atlassian.net/browse/OPERATOR-740)
2. They used different service types for different service, this should be fixed after Jiafeng merge the mambu request.

Glad that dry-run tool helped find several issues that were fixed in this PR too:
1. iks annotaiton
2. stork is enabled by default
3. use different service types for different service
4. when $usedrivesAndPartitions is false, useAll should be set.

### CODE REVIEW

File: community/portworx/values.yaml
Comment (pure-jliao) [inline community/portworx/values.yaml]: should we always avoid generating stc based on helm parameters here? feel it's hard to handle everything there.

Comment (haibinxie) [reply community/portworx/values.yaml]: good point, i actually just changed it to true. which means by default helm will generate the stc. I am only aware of one parameter that is not handled by operator generated stc, which is the is-iks annotation set to true.

Comment (pure-jliao) [reply community/portworx/values.yaml]: hmm I mean if we should only rely on operator to do the operation not helm. In case of helm deployment, do we expect the customers not to change the daemonset spec manually?

File: community/portworx/values.yaml
Comment (dsupure) [inline community/portworx/values.yaml:27]: Why we don't need `imagePullSecrets`?

File: community/portworx/values.yaml
Comment (dsupure) [inline community/portworx/values.yaml:3]: Do we need to enable the migration by default?

**Comment by dsupure** [comment]
@haibinxie : I assume this PR is from the `daemonset helm chart`? Should we keep it and have a seperate helm chart for the operator?

File: community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml
Comment (dsupure) [inline community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml:38]: Is it possible to run it in a for loop?

File: community/portworx/templates/serviceaccount-hook.yaml
Comment (dsupure) [inline community/portworx/templates/serviceaccount-hook.yaml:29]: Same comment in the previous PR..The permission seems too wide to me.

File: community/portworx/templates/storage-cluster.yaml
Comment (dsupure) [inline community/portworx/templates/storage-cluster.yaml]: Do we need EKS, PKS etc?

File: community/portworx/templates/storage-cluster.yaml
Comment (dsupure) [inline community/portworx/templates/storage-cluster.yaml]: Do we need AKS?

File: community/portworx/templates/storage-cluster.yaml
Comment (dsupure) [inline community/portworx/templates/storage-cluster.yaml]: We probably don't need this.

File: community/portworx/templates/portworx-operator.yaml
Comment (dsupure) [inline community/portworx/templates/portworx-operator.yaml]: nit: newline

File: community/portworx/values.yaml
Comment (haibinxie) [reply community/portworx/values.yaml:27]: it's a weird parameter in this chart, registry secret is used by px and stork, imagepullSecrets is used by other images. in operator we only support same secret for all images.

File: community/portworx/values.yaml
Comment (haibinxie) [reply community/portworx/values.yaml:3]: it's enabled by default.

File: community/portworx/templates/storage-cluster.yaml
Comment (haibinxie) [reply community/portworx/templates/storage-cluster.yaml]: not used, i will remove them

File: community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml
Comment (haibinxie) [reply community/portworx/templates/hooks/pre-upgrade/retain-daemonset-install.yaml:38]: we still need a list even with for loop

File: community/portworx/values.yaml
Comment (haibinxie) [reply community/portworx/values.yaml]: yes, customer should only give values.yaml, and we will generate stc. which is more reliable so far as i see.

File: community/portworx/templates/storage-cluster.yaml
Comment (pure-jliao) [inline community/portworx/templates/storage-cluster.yaml:46]: curious to know when migrate using helm, how does the migration work? do we just create a new stc using this template? that case how do we handle this flag?

**Review by pure-jliao — APPROVED** [review]

File: community/portworx/templates/storage-cluster.yaml
Comment (haibinxie) [reply community/portworx/templates/storage-cluster.yaml:46]: yes, we will create the stc here, then operator finds the stc exists and it will not create a new one. the migration thread sees the daemonset exists so it will wait for approval before starting migration

**Pull request merged**
